diff --git a/config.h b/config.h
index 744ef42..e3dd397 100644
--- a/config.h
+++ b/config.h
@@ -5,8 +5,8 @@
  *
  * font: see http://freedesktop.org/software/fontconfig/fontconfig-user.html
  */
-static char *font = "Liberation Mono:pixelsize=12:antialias=true:autohint=true";
-static int borderpx = 2;
+static char *font = "Recursive Mn Csl St:pixelsize=34";
+static int borderpx = 0;
 
 /*
  * What program is execed by st depends of these precedence rules:
@@ -26,7 +26,7 @@ char *stty_args = "stty raw pass8 nl -echo -iexten -cstopb 38400";
 char *vtiden = "\033[?6c";
 
 /* Kerning / character bounding-box multipliers */
-static float cwscale = 1.0;
+static float cwscale = 0.95;
 static float chscale = 1.0;
 
 /*
@@ -73,11 +73,11 @@ static unsigned int cursorthickness = 2;
  *    Bold affects lines thickness if boxdraw_bold is not 0. Italic is ignored.
  * 0: disable (render all U25XX glyphs normally from the font).
  */
-const int boxdraw = 0;
-const int boxdraw_bold = 0;
+const int boxdraw = 1;
+const int boxdraw_bold = 1;
 
 /* braille (U28XX):  1: render as adjacent "pixels",  0: use font */
-const int boxdraw_braille = 0;
+const int boxdraw_braille = 1;
 
 /*
  * bell volume. It must be a value between -100 and 100. Use 0 for disabling
@@ -93,7 +93,17 @@ static int bellvolume = 0;
  */
 static Envvar envvars[] = {
 	/* Name,	Value,		Override */
-	{ "EDITOR",	"/usr/bin/vim",	1 },
+#ifdef GRUVBOX_LIGHT
+	{ "BAT_THEME",	"gruvbox-light",	1 },
+#elif GRUVBOX_DARK
+	{ "BAT_THEME",	"gruvbox-dark",		1 },
+	{ "MY_NVIM_BG", "dark",			1 },
+#elif URXVT_LIGHT
+	{ "BAT_THEME",	"ansi",			1 },
+#else
+	{ "BAT_THEME",	"ansi",			1 },
+	{ "MY_NVIM_BG", "dark",			1 },
+#endif
 };
 
 /* default TERM value */
@@ -118,6 +128,91 @@ unsigned int tabspaces = 8;
 
 /* Terminal colors (16 first used in escape sequence) */
 static const char *colorname[] = {
+#if   defined GRUVBOX_LIGHT
+
+	/* 8 normal colors */
+	"#32302f",
+	"#ea6962",
+	"#a9b665",
+	"#d8a657",
+	"#7daea3",
+	"#d3869b",
+	"#89b482",
+	"#d4be98",
+
+	/* 8 bright colors */
+	"#32302f",
+	"#ea6962",
+	"#a9b665",
+	"#d8a657",
+	"#7daea3",
+	"#d3869b",
+	"#89b482",
+	"#d4be98",
+
+	[255] = 0,
+
+	/* more colors can be added after 255 to use with DefaultXX */
+	"#d4be98",
+	"#1d2021",
+
+#elif defined GRUVBOX_DARK
+
+	/* 8 normal colors */
+	"#654735",
+	"#c14a4a",
+	"#6c782e",
+	"#b47109",
+	"#45707a",
+	"#945e80",
+	"#4c7a5d",
+	"#f2e5bc",
+
+	/* 8 bright colors */
+	"#654735",
+	"#c14a4a",
+	"#6c782e",
+	"#b47109",
+	"#45707a",
+	"#945e80",
+	"#4c7a5d",
+	"#f2e5bc",
+
+	[255] = 0,
+
+	/* more colors can be added after 255 to use with DefaultXX */
+	"#654735",
+	"#f9f5d7",
+
+#elif defined URXVT_LIGHT
+
+	/* 8 normal colors */
+	"#000000",
+	"#a80000",
+	"#00a800",
+	"#a8a800",
+	"#0000a8",
+	"#a800a8",
+	"#00a8a8",
+	"#cfcfcf",
+
+	/* 8 bright colors */
+	"#000054",
+	"#ff0054",
+	"#00ff54",
+	"#ffff54",
+	"#0000ff",
+	"#ff00ff",
+	"#00ffff",
+	"#ffffff",
+
+	[255] = 0,
+
+	/* more colors can be added after 255 to use with DefaultXX */
+	"#000000",
+	"#FFFFFF",
+
+#else
 	/* 8 normal colors */
 	"black",
 	"red3",
@@ -143,6 +238,7 @@ static const char *colorname[] = {
 	/* more colors can be added after 255 to use with DefaultXX */
 	"#cccccc",
 	"#555555",
+#endif
 };
 
 
@@ -150,8 +246,13 @@ static const char *colorname[] = {
  * Default colors (colorname index)
  * foreground, background, cursor, reverse cursor
  */
+#if defined GRUVBOX_LIGHT || defined GRUVBOX_DARK || defined URXVT_LIGHT
+unsigned int defaultfg = 256;
+unsigned int defaultbg = 257;
+#else
 unsigned int defaultfg = 7;
 unsigned int defaultbg = 0;
+#endif
 static unsigned int defaultcs = 256;
 static unsigned int defaultrcs = 257;
 
@@ -197,7 +298,12 @@ static uint forcemousemod = ShiftMask;
  */
 static MouseShortcut mshortcuts[] = {
 	/* mask                 button   function        argument       release */
-	{ XK_ANY_MOD,           Button2, selpaste,       {.i = 0},      1 },
+
+	/* Mouse scrolling support */
+	{ XK_ANY_MOD,           Button4, kscrollup,      {.i = 4},      0, /* !alt */ -1 },
+	{ XK_ANY_MOD,           Button5, kscrolldown,    {.i = 4},      0, /* !alt */ -1 },
+	/* END Mouse scrolling support */
+
 	{ ShiftMask,            Button4, ttysend,        {.s = "\033[5;2~"} },
 	{ XK_ANY_MOD,           Button4, ttysend,        {.s = "\031"} },
 	{ ShiftMask,            Button5, ttysend,        {.s = "\033[6;2~"} },
@@ -207,6 +313,7 @@ static MouseShortcut mshortcuts[] = {
 /* Internal keyboard shortcuts. */
 #define MODKEY Mod1Mask
 #define TERMMOD (ControlMask|ShiftMask)
+#define CLIPMOD (ControlMask|Mod1Mask)
 
 static Shortcut shortcuts[] = {
 	/* mask                 keysym          function        argument */
@@ -214,16 +321,26 @@ static Shortcut shortcuts[] = {
 	{ ControlMask,          XK_Print,       toggleprinter,  {.i =  0} },
 	{ ShiftMask,            XK_Print,       printscreen,    {.i =  0} },
 	{ XK_ANY_MOD,           XK_Print,       printsel,       {.i =  0} },
-	{ TERMMOD,              XK_Prior,       zoom,           {.f = +1} },
-	{ TERMMOD,              XK_Next,        zoom,           {.f = -1} },
+	{ TERMMOD,              XK_plus,        zoom,           {.f = +1} },
+	{ TERMMOD,              XK_underscore,  zoom,           {.f = -1} },
 	{ TERMMOD,              XK_Home,        zoomreset,      {.f =  0} },
 	{ TERMMOD,              XK_C,           clipcopy,       {.i =  0} },
 	{ TERMMOD,              XK_V,           clippaste,      {.i =  0} },
+	{ CLIPMOD,              XK_c,           clipcopy,       {.i =  0} },
+	{ CLIPMOD,              XK_v,           clippaste,      {.i =  0} },
+	/*
+	 * NOTE: Note the change in case from XK_C to XK_c. That's because
+	 * our mapping doesn't have ShiftMask. i.e. We don't have the Shift
+	 * key pressed. So, we will input a lowercase c (i.e. XK_c) instead
+	 * of and uppercase C (i.e. XK_C)
+	 */
 	{ TERMMOD,              XK_Y,           selpaste,       {.i =  0} },
 	{ ShiftMask,            XK_Insert,      selpaste,       {.i =  0} },
 	{ TERMMOD,              XK_Num_Lock,    numlock,        {.i =  0} },
-	{ ShiftMask,            XK_Page_Up,     kscrollup,      {.i = -1} },
-	{ ShiftMask,            XK_Page_Down,   kscrolldown,    {.i = -1} },
+	{ ShiftMask,            XK_Prior,       kscrollup,      {.i = -1} },
+	{ ShiftMask,            XK_Next,        kscrolldown,    {.i = -1} },
+	{ ShiftMask,            XK_Up,          kscrollup,      {.i =  1} },
+	{ ShiftMask,            XK_Down,        kscrolldown,    {.i =  1} },
 };
 
 /*
